Add documentation on how to run VTAdmin in VTop (#1073)
vitessio/website: content/en/docs/14.0/reference/vtadmin/running_vtadmin_in_vtop.md

---

> title: Running VTAdmin in VTop

Reviewer: "running **vtadmin** in VTop" is redundant when this is under `../../vtadmin`, i'd rename this file (and title) to just "Running in VTop" (maybe "Running _with_ ..." ?

Author: Yes, that is a good point

Author: Fixed

Author: Okay changing it

description: How to configure VTop to run VTAdmin
---

## Get Started

> VTAdmin only runs in a single Vitess cluster configuration in the Vitess operator.

Reviewer: this should be in `<{{ info }}>`

Reviewer: also probably a note about how we're working on this (we are, right?)

Author: I don't think that is going to change since the Vitess spec is one cluster. Currently you can't have multiple clusters in a specification. At least that is what my understanding is

Author: Why does it need to be in `<{{info}}>`? Should I remove the getting started header then, since there are only 2 lines in it

Reviewer: I mean this specific bit about only running in a single-cluster. VTAdmin advertises itself as multi-cluster, so the fact this is not true for the VTOp version should warrant special callout

Author: Fixed

Please also read the [Operator's Guide](../operators_guide) to learn more about configurations available and how to use them.

## Compatibility

> The support for deploying VTAdmin in VTop has been added in [release 2.7.1](https://github.com/planetscale/vitess-operator/releases/tag/v2.7.1) onwards.

Author: Currently this link is broken, but will work after a new release of the operator today.

Author: This link works now

> The support for deploying VTAdmin in VTop has been added in [release 2.7.1](https://github.com/planetscale/vitess-operator/releases/tag/v2.7.1) onwards.

Reviewer: ```suggestion
Support for deploying VTAdmin in VTop has been added in [release 2.7.1](https://github.com/planetscale/vitess-operator/releases/tag/v2.7.1) onwards.
```

Author: Fixed

## Overview

> VTop deploys VTAdmin in 2 separate containers running on the same pod. One for running the `vtadmin-api` and one for `vtadmin-web`. Please look at the [Architecture](../architecture) for more details on how they interact with each other.

Reviewer: ```suggestion
VTop deploys VTAdmin in 2 separate containers running on the same pod. One for running the `vtadmin-api` and one for `vtadmin-web`. Please look at the [architecture docs](../architecture) for details on how they interact with each other.
```

Author: Fixed

> Services are created on top of both `vtadmin-api` and `vtadmin-web` which can be used to access them after either port-forwarding or assigning an external IP address to the service.

Reviewer: ```suggestion
VTop then creates services on top of both `vtadmin-api` and `vtadmin-web`, which can be used to access them after either port-forwarding or assigning an external IP address to the service.
```

Author: Fixed

> VTop creates the `discovery.json` file automatically which is needed to connect to vtctld and vtgate services. It connects to the global vtctld service and per-cell service of vtgates, both of which VTop creates automatically. So no configuration from the users is required for discovering these components.

Reviewer: ```suggestion
Finally, VTop creates the `discovery.json` file automatically which is needed to connect to `vtctld` and `vtgate` services. It connects to the global `vtctld` service and per-cell service of `vtgates`, both of which VTop creates automatically. No configuration from the users is required for discovering these components.
```

Author: Fixed

> ## Configuration of VTAdmin in VTop

Reviewer: ```suggestion
## Configuring VTAdmin in VTop
```

Author: Fixed

> VTAdmin specification in the cluster specification lives at the same level as the vtctld specification. 

Reviewer: I found this a bit confusing to parse, but I'm also not a k8s expert so, if my suggestion is technically wrong at all please reject it!

```suggestion
The VTAdmin configuration section lives at the same level as the vtctld configuration in the cluster specification.
```

Reviewer: (in either case, you've got trailing whitespace, apparently, which i only noticed in my suggestion diff 😂)

Author: This is perfectly correct.

Author: Fixed

The following options are available for configuring VTAdmin - 

> - `Rbac` is a secret source. It is the role based access control rules to use for VTAdmin API. More information on role based access control can be found [here](../role-based-access-control).

Reviewer: Some inconsistencies with the style of the other docs for these terms here:

```suggestion
- `RBAC` is a secret source. It is the role-based access control rules to use for VTAdmin API. More information on role-based access control can be found [here](../role-based-access-control).
```

Author: Fixed

- `APIAddresses` as a list of strings. Since the VTAdmin web UI runs on the client side, it needs the API address to use to query the API. We can't use the internal kubernetes service address or the ip of the pod, since they aren't visible outside the cluster. The API Address must be provided by the user based on how they export the service of VTAdmin API. In our example configuration (see below) we port-forward the service to port `14001` and therefore that is the address provided there. This value is a list because the address to be used for VTAdmin web in each cell might be different. If only 1 value is provided then, that is used for all the cells. The ideal way to deploy this would be to export each individual VTAdmin service (that we create) in each cell and attach external IPs to them and provide those IPs here.
- `Replicas` - Number of VTAdmin deployments required per cell. We setup a service on top of the web and API connections, so load-balancing comes out of the box.
- `WebResources` - Resource requests and limits for the container running the VTAdmin web server.

> - `APIResources` - Resource requests and limits for the container running the VTAdmin API server.

Reviewer: should these by styled as "running the `vtdamin-{web, api}` server" or as-written?

Author: I am fine with either. I have been using canonically.

> - `ReadOnly` - Configuration to set the VTAdmin web UI to be read only.

Reviewer: ```suggestion
- `ReadOnly` - Configuration to set the VTAdmin web UI to be read-only.
```

Author: Fixed
